### PR TITLE
add date and time options to content rotation

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

File: molo/core/tasks.py [hunk 1]
```diff
@@ -26,24 +26,32 @@ def rotate_content():
...
-        random_article.save_revision().publish()
-
-        article = main.latest_articles().last()
-        article.featured_in_latest = False
-        article.save_revision().publish()
+        rotate_latest(main_lang, index, main, site_settings, days, day)
+        rotate_featured_in_homepage(main_lang)
+
+
+def rotate_latest(main_lang, index, main, site_settings, days, day):
```
Comment (KaitCrawford) [inline molo/core/tasks.py]: Please add some comments in this method to explain things

File: molo/core/tests/test_tasks.py [hunk 2]
```diff
@@ -28,42 +28,144 @@ def setUp(self):
...
         self.assertEquals(
-            first_article_old, self.main.latest_articles()[1].pk)
+            first_article_old, self.main.latest_articles()[2].pk)
         self.assertNotEquals(
-            last_article_old, self.main.latest_articles()[9].pk)
-        self.assertEquals(
-            second_last_article_old, self.main.latest_articles()[9].pk)
+            last_article_old, self.main.latest_articles()[8].pk)
+
+    def test_latest_rotation_no_days(self):
```
Comment (KaitCrawford) [inline molo/core/tests/test_tasks.py]: Please add some comments to this test

File: molo/core/tests/test_tasks.py [hunk 3]
```diff
@@ -28,42 +28,144 @@ def setUp(self):
...
+        self.mk_articles(self.yourmind_sub, count=10, featured_in_latest=True)
+        self.mk_articles(self.yourmind_sub, count=10, featured_in_latest=False)
+        self.assertEquals(self.main.latest_articles().count(), 10)
+        first_article_old = self.main.latest_articles()[0].pk
+        last_article_old = self.main.latest_articles()[9].pk
+        rotate_content()
+        self.assertEquals(first_article_old, self.main.latest_articles()[0].pk)
+        self.assertEquals(last_article_old, self.main.latest_articles()[9].pk)
+
+    def test_latest_rotation_no_time(self):
```
Comment (KaitCrawford) [inline molo/core/tests/test_tasks.py]: Please add some comments to this test

File: molo/core/tests/test_tasks.py [hunk 4]
```diff
@@ -28,42 +28,144 @@ def setUp(self):
...
+        self.mk_articles(self.yourmind_sub, count=10, featured_in_latest=True)
+        self.mk_articles(self.yourmind_sub, count=10, featured_in_latest=False)
+        self.assertEquals(self.main.latest_articles().count(), 10)
+        first_article_old = self.main.latest_articles()[0].pk
+        last_article_old = self.main.latest_articles()[9].pk
+        rotate_content()
+        self.assertEquals(first_article_old, self.main.latest_articles()[0].pk)
+        self.assertEquals(last_article_old, self.main.latest_articles()[9].pk)
+
+    def test_latest_rotation_no_start_or_end_date(self):
```
Comment (KaitCrawford) [inline molo/core/tests/test_tasks.py]: Please add some comments to this test

**Comment by KaitCrawford** [comment]
@codiebeulaine left some requests for comments mostly because I don't understand what's going on in the relevant code. Your tests specifically.

File: molo/core/tasks.py [hunk 5]
```diff
@@ -12,38 +12,60 @@
...
+    featured_in_latest. It checks whether current date falls within the set
+    date range for content rotation. It then checks whether the current weekday
+    is set to rotate, and then rotates an articles for each hour the admin has
+    set."""
+    if site_settings.content_rotation_start_date < timezone.now() \
+            < site_settings.content_rotation_end_date:
+        # checks if the current weekday is set to rotate
+        if days[day - 1]:
+            for time in site_settings.time:
+                if str(time)[:2] == str(datetime.now().hour):
```
Comment (miltontony) [inline molo/core/tasks.py]: depending on the value of `time`, you should be able to get to the `hour` without using `str`

**Comment by codiebeulaine** [comment]
screen shot 2016-08-05 at 3 04 11 pm

**Comment by codiebeulaine** [comment]
@miltontony ready for review

**Comment by miltontony** [comment]
👍 🍰

**Pull request merged**